src/Request/RequestHandler.php

```diff
@@ -317,4 +352,96 @@ public function request($httpMethod, $service, $parameters = [ ])
             $parameters
         );
     }
+
+    /**
+     * Generates oauth signature
+     */
+    public function generateOauthSignature(string $method, string $path, array $parameters = [])
+    {
+        switch ($this->signatureMethod) {
+            case 'RSA-SHA1':
+                return $this->generateRSASHA1Signature($method, $path, $parameters);
+                break;
+            case 'HMAC-SHA1':
+                return 'stub data';
+                break;
+            default:
+                throw new ConfigException(ConfigException::UNKNOWN_SIGNATURE_METHOD, $this->signatureMethod);
+        }
+    }
+
+    protected function generateRSASHA1Signature(string $method, string $path, array $parameters)
+    {
+        if (!file_exists($this->privateKey)) {
+            throw new ConfigException(ConfigException::PRIVATE_KEY_NOT_FOUND, $this->privateKey);
+        }
+
+        $file = fopen($this->privateKey, 'r');
+        $contents = fread($file, 8192);
+        fclose($file);
+
+        $privateKey = openssl_pkey_get_private($contents);
+        if ($privateKey === false) {
+            throw new ConfigException(ConfigException::PRIVATE_KEY_INVALID, $this->privateKey);
+        }
+
+        $sbs = sprintf(
+            '%s&%s&%s',
+            $method,
+            $this->oauthEscape($this->endpoint . '/' . $path),
+            $this->oauthEscape($this->sortParameters($parameters))
+        );
+
+        openssl_sign($sbs, $signature, $privateKey);
+
+        openssl_free_key($privateKey);
+
+        return base64_encode($signature);
+    }
+
+    /**
+     * Sorts query parameters
```

Comment:
Can you give a little more information in this comment about what this does and why it does it?

It's not immediately clear to me.

Thanks